packages/core/src/lib/resolver.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const path = require('path');
+
+/**
+ * @func resolveFileInPackage
+ * Resolves a file inside a package
+ */
+const resolveFileInPackage = (packageName, ...pathElements) => {
+  return require.resolve(path.join(packageName, ...pathElements));
+};
+
+/**
+ * @func resolveDirInPackage
+ * Resolves a file inside a package
+ */
+const resolveDirInPackage = (packageName, ...pathElements) => {
+  return path.dirname(resolveFileInPackage(packageName, ...pathElements));
+};
+
+/**
+ * @func resolvePackageFolder
+ * Resolves the location of a package on disc
+ */
+const resolvePackageFolder = packageName => {
+  return path.dirname(resolveFileInPackage(packageName, 'package.json'));
+};
```

Comment:
Just a question for this. The whole concept is working without process.cwd() is that correct?
When it tries to resolve the given files/directories it will walk from the resolver location through the filesystem or where would it start?

Reply:
Patternlab loads files from installed NodeJS packages. It starts loading via whatever resolver a package manager could configure inside the Node runtime. In my case, I want to support Yarn v2 with PnP active eventually. In such a setup, no `node_modules` folder is used at all and every package is stored as a zip file. Here is an example how files inside a package are to be found:

https://yarnpkg.com/features/pnp#packages-are-stored-inside-zip-archives-how-can-i-access-their-files

Reply:
If it does not interfere with other resolvers like npm or yarn v1 everything is fine.

Reply:
@JosefBredereck if uikit resolution works via `require.resolve`, the updated code *ought* to work under any other setup. If it doesn't it is a bug and needs to be fixed, which I'm glad to provide a fix for.